src/codegen/constant-propagation.lisp

```diff
@@ -0,0 +1,129 @@
+(defpackage #:coalton-impl/codegen/constant-propagation
+  (:use
+   #:cl
+   #:coalton-impl/codegen/ast)
+  (:local-nicknames
+   (#:tc #:coalton-impl/typechecker))
+  (:import-from
+   #:coalton-impl/codegen/traverse
+   #:action
+   #:count-applications
+   #:count-nodes
+   #:make-traverse-let-action-skipping-cons-bindings
+   #:*traverse*
+   #:traverse
+   #:traverse-with-binding-list)
+  (:export
+   #:propagate-constants))
+
+(in-package #:coalton-impl/codegen/constant-propagation)
+
+(defvar *constant-bindings*)
+(setf (documentation '*constant-bindings* 'variable)
+      "Bound inside CONSTANT-PROPAGATION.")
+
+(defun constant-var-value (var &key no-error)
+  (the (or node null)
+       (or (cdr (assoc var *constant-bindings*))
```

Comment:
I know it's more typing but can you lift this out into a let+cond?

```
(let ((val ...))
  (cond
    (val ..)
    (no-error ..)
    (t ..)))
```

Reply:
Committed in https://github.com/coalton-lang/coalton/pull/1212/commits/63a82915937c7eb77faa3429b4fd9e3bcd710a9e.